src/helm/proxy/clients/vision_language/idefics_client.py

```diff
@@ -0,0 +1,197 @@
+
+    def make_request(self, request: Request) -> RequestResult:
+        assert request.model in _models, f"Not a valid model for this client: {request.model}"
+        assert request.multimodal_prompt is not None, "Multimodal prompt is required"
+
+        loaded_model_processor: LoadedIDEFICSModelProcessor = self._get_model(request.model)
+        model = loaded_model_processor.model
+        processor = loaded_model_processor.processor
+
+        input_args: Dict[str, Union[str, bool]] = {"return_tensors": "pt"}
+        generation_args = {
+            "max_length": request.max_tokens,
+            "bad_words_ids": processor.tokenizer(self.BAD_WORD_TOKENS, add_special_tokens=False).input_ids,
+        }
+
+        if self.END_OF_UTTERANCE_TOKEN in request.stop_sequences:
+            # Following https://huggingface.co/HuggingFaceM4/idefics-80b-instruct,
+            # specify <end_of_utterance> as an exit condition.
+            input_args["add_end_of_utterance_token"] = False
+            exit_condition = processor.tokenizer(self.END_OF_UTTERANCE_TOKEN, add_special_tokens=False).input_ids
+            generation_args["eos_token_id"] = exit_condition
+
+        multimodal_prompt: List[Union[str, Image]] = [
+            open_image(media_object.location)
+            if media_object.is_type("image") and media_object.location
+            else media_object.text
+            for media_object in request.multimodal_prompt.media_objects
+        ]
+        prompt_text: str = request.multimodal_prompt.text.replace(self.END_OF_UTTERANCE_TOKEN, " ")
+
+        try:
+
+            def do_it():
+                inputs = processor(multimodal_prompt, **input_args).to(self._device)
+                generated_ids = model.generate(**inputs, **generation_args)
+                generated_text: str = processor.batch_decode(generated_ids, skip_special_tokens=True)[0]
+                assert generated_text.startswith(
+                    prompt_text
+                ), f"Generated text: {generated_text} does not start with prompt: {prompt_text}"
+
+                # Remove the prompt from the generated text
+                generated_text = generated_text[len(prompt_text) :].strip()
+                return {"output": generated_text}
+
+            # Include the prompt and model name in the cache key
+            cache_key = Client.make_cache_key(
+                raw_request={
+                    "model": request.model,
+                    "prompt": generate_uid_for_multimodal_prompt(request.multimodal_prompt),
+                    **generation_args,
+                },
+                request=request,
+            )
+            result, cached = self._cache.get(cache_key, wrap_request_time(do_it))
+        except RuntimeError as e:
+            return RequestResult(success=False, cached=False, error=str(e), completions=[], embedding=[])
+
+        completions: List[Sequence] = [Sequence(text=result["output"], logprob=0, tokens=[])]
```

Comment:
Does it cause problems that `tokens` is empty?

Reply:
I'm not sure how to get the logprobs so I left it empty for now. I will at least tokenize and return tokens then.